File: backend-php/dao/ReclamationDAO.php
<?php
/**
 * DAO : accès à la table reclamations.
 * Équivalent PHP de com.ihecride.dao.ReclamationDAO.
 */

require_once __DIR__ . '/../utils/DBConnection.php';
require_once __DIR__ . '/../models/Reclamation.php';

class ReclamationDAO {

    public function insert(Reclamation $r): int {
        $pdo = DBConnection::getConnection();
        try {
            $stmt = $pdo->prepare(
                "INSERT INTO reclamations(declarant_id, cible_id, type, description) VALUES(?,?,?,?)"
            );
            $ok = $stmt->execute([
                $r->declarantId,
                $r->cibleId,
                $r->type,
                $r->description
            ]);
            if ($ok && $stmt->rowCount() > 0) return (int) $pdo->lastInsertId();
        } catch (PDOException $e) {
            error_log('ReclamationDAO::insert ' . $e->getMessage());
        }
        return -1;
    }

    /** @return Reclamation[] */
    public function findAll(): array {
        $sql = "SELECT r.*, CONCAT(u.prenom,' ',u.nom) AS declarant_nom
                FROM reclamations r JOIN users u ON r.declarant_id=u.id
                ORDER BY date_soumission DESC";
        $stmt = DBConnection::getConnection()->query($sql);
        $list = [];
        foreach ($stmt->fetchAll() as $row) $list[] = $this->map($row);
        return $list;
    }

    /** @return Reclamation[] */
    public function findByDeclarant(int $declarantId): array {
        $sql = "SELECT r.*, CONCAT(u.prenom,' ',u.nom) AS declarant_nom
                FROM reclamations r JOIN users u ON r.declarant_id=u.id
                WHERE declarant_id = ? ORDER BY date_soumission DESC";
        $stmt = DBConnection::getConnection()->prepare($sql);
        $stmt->execute([$declarantId]);
        $list = [];
        foreach ($stmt->fetchAll() as $row) $list[] = $this->map($row);
        return $list;
    }

    public function updateStatut(int $id, string $statut): bool {
        $stmt = DBConnection::getConnection()->prepare("UPDATE reclamations SET statut = ? WHERE id = ?");
        return $stmt->execute([$statut, $id]) && $stmt->rowCount() > 0;
    }

    private function map(array $rs): Reclamation {
        $r = new Reclamation();
        $r->id              = isset($rs['id']) ? (int) $rs['id'] : null;
        $r->declarantId     = isset($rs['declarant_id']) ? (int) $rs['declarant_id'] : null;
        $r->cibleId         = (isset($rs['cible_id']) && $rs['cible_id'] !== null) ? (int) $rs['cible_id'] : null;
        $r->declarantNom    = $rs['declarant_nom'] ?? null;
        $r->type            = $rs['type'] ?? null;
        $r->description     = $rs['description'] ?? null;
        $r->statut          = $rs['statut'] ?? null;
        $r->dateSoumission  = $rs['date_soumission'] ?? null;
        return $r;
    }
}

File: backend/src/main/java/com/ihecride/dao/ReclamationDAO.java
package com.ihecride.dao;

import com.ihecride.models.Reclamation;
import com.ihecride.utils.DBConnection;

import java.sql.*;
import java.util.ArrayList;
import java.util.List;

public class ReclamationDAO {

    public int insert(Reclamation r) {
        String sql = "INSERT INTO reclamations(declarant_id, cible_id, type, description) VALUES(?,?,?,?)";
        try (Connection c = DBConnection.getConnection();
             PreparedStatement ps = c.prepareStatement(sql, Statement.RETURN_GENERATED_KEYS)) {
            ps.setInt(1, r.getDeclarantId());
            if (r.getCibleId() != null) ps.setInt(2, r.getCibleId()); else ps.setNull(2, Types.INTEGER);
            ps.setString(3, r.getType());
            ps.setString(4, r.getDescription());
            if (ps.executeUpdate() > 0) {
                try (ResultSet k = ps.getGeneratedKeys()) {
                    if (k.next()) return k.getInt(1);
                }
            }
        } catch (SQLException e) { e.printStackTrace(); }
        return -1;
    }

    public List<Reclamation> findAll() {
        List<Reclamation> list = new ArrayList<>();
        String sql = "SELECT r.*, CONCAT(u.prenom,' ',u.nom) AS declarant_nom FROM reclamations r JOIN users u ON r.declarant_id=u.id ORDER BY date_soumission DESC";
        try (Connection c = DBConnection.getConnection();
             Statement st = c.createStatement();
             ResultSet rs = st.executeQuery(sql)) {
            while (rs.next()) list.add(map(rs));
        } catch (SQLException e) { e.printStackTrace(); }
        return list;
    }

    public List<Reclamation> findByDeclarant(int declarantId) {
        List<Reclamation> list = new ArrayList<>();
        String sql = "SELECT r.*, CONCAT(u.prenom,' ',u.nom) AS declarant_nom FROM reclamations r JOIN users u ON r.declarant_id=u.id WHERE declarant_id = ? ORDER BY date_soumission DESC";
        try (Connection c = DBConnection.getConnection();
             PreparedStatement ps = c.prepareStatement(sql)) {
            ps.setInt(1, declarantId);
            try (ResultSet rs = ps.executeQuery()) {
                while (rs.next()) list.add(map(rs));
            }
        } catch (SQLException e) { e.printStackTrace(); }
        return list;
    }

    public boolean updateStatut(int id, String statut) {
        String sql = "UPDATE reclamations SET statut = ? WHERE id = ?";
        try (Connection c = DBConnection.getConnection();
             PreparedStatement ps = c.prepareStatement(sql)) {
            ps.setString(1, statut);
            ps.setInt(2, id);
            return ps.executeUpdate() > 0;
        } catch (SQLException e) { e.printStackTrace(); }
        return false;
    }

    private Reclamation map(ResultSet rs) throws SQLException {
        Reclamation r = new Reclamation();
        r.setId(rs.getInt("id"));
        r.setDeclarantId(rs.getInt("declarant_id"));
        int cid = rs.getInt("cible_id");
        if (!rs.wasNull()) r.setCibleId(cid);
        try { r.setDeclarantNom(rs.getString("declarant_nom")); } catch (Exception ignored) {}
        r.setType(rs.getString("type"));
        r.setDescription(rs.getString("description"));
        r.setStatut(rs.getString("statut"));
        r.setDateSoumission(rs.getTimestamp("date_soumission"));
        return r;
    }
}
